oak-commons/src/main/java/org/apache/jackrabbit/oak/commons/conditions/Checks.java

```diff
@@ -0,0 +1,118 @@
+package org.apache.jackrabbit.oak.commons.conditions;
+
+import java.util.Objects;
+
+import org.apache.jackrabbit.oak.commons.properties.SystemPropertySupplier;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public final class Checks {
+
+    private Checks() {
+        // no instances for you
+    }
+
+    private static Logger LOG = LoggerFactory.getLogger(Checks.class);
+
+    // when true, message template are arguments checked even when the condition
+    // to check for is true
+    private static boolean CHECKMESSAGETEMPLATE = SystemPropertySupplier
+            .create("oak.precondition.checks.CHECKMESSAGETEMPLATE", false).loggingTo(LOG).get();
+
+    /**
+     * Checks the specified expression
+     * 
+     * @param expression
+     *            to check
+     * @throws IllegalArgumentException
+     *             when false
+     */
+    public static void checkArgument(boolean expression) throws IllegalArgumentException {
+        if (!expression) {
+            throw new IllegalArgumentException();
+        }
+    }
+
+    /**
+     * Checks the specified expression
+     * 
+     * @param expression
+     *            to check
+     * @param message
+     *            to use in exception
+     * @throws IllegalArgumentException
+     *             when false
+     */
+    public static void checkArgument(boolean expression, @NotNull String message) throws IllegalArgumentException {
+
+        Objects.requireNonNull(message);
+
+        if (!expression) {
+            throw new IllegalArgumentException(message);
+        }
+    }
+
+    /**
+     * Checks the specified expression
+     * 
+     * @param expression
+     *            to checks
+     * @param messageTemplate
+     *            to use in exception (using {@link String#format} syntax
+     * @throws IllegalArgumentException
+     *             when false
+     */
+    public static void checkArgument(boolean expression, @NotNull String messageTemplate, @Nullable Object... messageArgs) {
+
+        Objects.requireNonNull(messageTemplate);
+
+        if (CHECKMESSAGETEMPLATE) {
+            checkTemplate(messageTemplate, messageArgs);
+        }
+
+        if (!expression) {
+            if (!CHECKMESSAGETEMPLATE) {
+                checkTemplate(messageTemplate, messageArgs);
+            }
+
+            String message = String.format(messageTemplate, messageArgs);
+            throw new IllegalArgumentException(message);
+        }
+    }
+
+    protected static boolean checkTemplate(@NotNull String messageTemplate, @Nullable Object... messageArgs) {
```

Comment:
the scope should be `default`.
Since it is final and cannot be extended, it doesn't make sense to use a `protected` scope.